src/Router/Router.js

```diff
@@ -0,0 +1,73 @@
+export default class Router {
+  constructor(routes) {
+    this.routes = routes;
+    this.container = document.querySelector('.container');
+    this.footer = document.querySelector('footer');
+  }
+
+  start() {
+    window.addEventListener('hashchange', () => {
+      this.hasChanged(this, this.routes);
+    });
+
+    this.hasChanged(this, this.routes);
+  }
+
+  hasChanged() {
+    if (window.location.hash.length > 0) {
+      for (let i = 0; i < this.routes.length; i++) {
+        const route = this.routes[i];
+        if (route.isActiveRoute(window.location.hash)) {
+          this.goToRoute(route.html, route.getParams(window.location.hash));
+          break;
+        }
+      }
+    } else {
+      for (let i = 0; i < this.routes.length; i++) {
+        const route = this.routes[i];
+        if (route.default) {
+          this.goToRoute(route.html);
+          break;
+        }
+      }
+    }
+  }
```

Comment:
Way to minimize, you use equal loops here
```suggestion
    for (let i = 0; i < this.routes.length; i++) {
      const route = this.routes[i];
        if (window.location.hash.length > 0) {
          if (route.isActiveRoute(window.location.hash)) {
            this.goToRoute(route.html, route.getParams(window.location.hash));
            break;
          }
        } else {
          if (route.default) {
            this.goToRoute(route.html);
            break;
          }
        }
      }
    }
```

Reply:
I had two solutions and chose the first. 
The first solution has a lot of code, but iterates over the values after the condition is met.
The second solution has less code, but it iterates over almost all values at the beginning.
I settled on the first solution.